pkg/storage/permanent/permanent.go

```diff
@@ -1,63 +1,60 @@
 package permanent
 
 import (
-	"os"
-
 	"github.com/iotaledger/hive.go/kvstore"
 	"github.com/iotaledger/hive.go/lo"
-	"github.com/iotaledger/iota-core/pkg/database"
+	"github.com/iotaledger/hive.go/runtime/ioutils"
+	"github.com/iotaledger/iota-core/pkg/storage/database"
 	"github.com/iotaledger/iota-core/pkg/storage/utils"
 )
 
 const (
-	unspentOutputsPrefix byte = iota
-	unspentOutputIDsPrefix
-	consensusWeightsPrefix
-	attestationsPrefix
-	throughputQuotaPrefix
+	sybilProtectionPrefix byte = iota
 )
 
 type Permanent struct {
-	Settings       *Settings
-	Commitments    *Commitments
-	UnspentOutputs kvstore.KVStore
-
-	unspentOutputIDs kvstore.KVStore
-	attestations     kvstore.KVStore
-	sybilProtection  kvstore.KVStore
-	throughputQuota  kvstore.KVStore
+	dbConfig           database.Config
+	store              kvstore.KVStore
+	storeHealthTracker *kvstore.StoreHealthTracker
+
+	settings    *Settings
+	commitments *Commitments
+
+	sybilProtection kvstore.KVStore
 }
 
 // New returns a new permanent storage instance.
-func New(dir *utils.Directory, db *database.Manager) (p *Permanent) {
-	return &Permanent{
-		Settings:       NewSettings(dir.Path("settings.bin")),
-		Commitments:    NewCommitments(dir.Path("commitments.bin")),
-		UnspentOutputs: lo.PanicOnErr(db.PermanentStorage().WithExtendedRealm([]byte{unspentOutputsPrefix})),
-
-		unspentOutputIDs: lo.PanicOnErr(db.PermanentStorage().WithExtendedRealm([]byte{unspentOutputIDsPrefix})),
-		attestations:     lo.PanicOnErr(db.PermanentStorage().WithExtendedRealm([]byte{attestationsPrefix})),
-		sybilProtection:  lo.PanicOnErr(db.PermanentStorage().WithExtendedRealm([]byte{consensusWeightsPrefix})),
-		throughputQuota:  lo.PanicOnErr(db.PermanentStorage().WithExtendedRealm([]byte{throughputQuotaPrefix})),
+func New(baseDir *utils.Directory, dbConfig database.Config) *Permanent {
+	store, err := database.StoreWithDefaultSettings(dbConfig.Directory, true, dbConfig.Engine)
+	if err != nil {
+		panic(err)
 	}
-}
 
-// UnspentOutputIDs returns the "unspent outputs ids" storage (or a specialized sub-storage if a realm is provided).
-func (p *Permanent) UnspentOutputIDs(optRealm ...byte) kvstore.KVStore {
-	if len(optRealm) == 0 {
-		return p.unspentOutputIDs
+	// TODO: What do we do if the DB is corrupted?
```

Comment:
Exit with a corresponding message telling the user to delete the database and start from scratch